Bug 1578363 - Add Microblaze support

[Gustavo: update for nspr 4.10.9]
Signed-off-by: Spenser Gilliland <spenser@gillilanding.com>
[Fabrice: update for nspr 4.20]
Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
[Giulio: update for nspr 4.22]
Signed-off-by: Giulio Benetti <giulio.benetti@micronovasrl.com>

diff -r 8487beb203e9 -r ab7d5c504daf nspr/pr/include/md/_linux.cfg
--- a/nspr/pr/include/md/_linux.cfg	Tue Sep 03 11:25:11 2019 +0200
+++ b/nspr/pr/include/md/_linux.cfg	Tue Sep 03 11:33:15 2019 +0200
@@ -1202,6 +1202,56 @@
 #define PR_BYTES_PER_WORD_LOG2   2
 #define PR_BYTES_PER_DWORD_LOG2  3
 
+#elif defined(__microblaze__)
+
+#if defined(__BIG_ENDIAN__)
+#define IS_BIG_ENDIAN 1
+#undef  IS_LITTLE_ENDIAN
+#else
+#define IS_LITTLE_ENDIAN 1
+#undef  IS_BIG_ENDIAN
+#endif
+
+#define PR_BYTES_PER_BYTE   1
+#define PR_BYTES_PER_SHORT  2
+#define PR_BYTES_PER_INT    4
+#define PR_BYTES_PER_INT64  8
+#define PR_BYTES_PER_LONG   4
+#define PR_BYTES_PER_FLOAT  4
+#define PR_BYTES_PER_DOUBLE 8
+#define PR_BYTES_PER_WORD   4
+#define PR_BYTES_PER_DWORD  8
+
+#define PR_BITS_PER_BYTE    8
+#define PR_BITS_PER_SHORT   16
+#define PR_BITS_PER_INT     32
+#define PR_BITS_PER_INT64   64
+#define PR_BITS_PER_LONG    32
+#define PR_BITS_PER_FLOAT   32
+#define PR_BITS_PER_DOUBLE  64
+#define PR_BITS_PER_WORD    32
+
+#define PR_BITS_PER_BYTE_LOG2   3
+#define PR_BITS_PER_SHORT_LOG2  4
+#define PR_BITS_PER_INT_LOG2    5
+#define PR_BITS_PER_INT64_LOG2  6
+#define PR_BITS_PER_LONG_LOG2   5
+#define PR_BITS_PER_FLOAT_LOG2  5
+#define PR_BITS_PER_DOUBLE_LOG2 6
+#define PR_BITS_PER_WORD_LOG2   5
+
+#define PR_ALIGN_OF_SHORT   2
+#define PR_ALIGN_OF_INT     4
+#define PR_ALIGN_OF_LONG    4
+#define PR_ALIGN_OF_INT64   4
+#define PR_ALIGN_OF_FLOAT   4
+#define PR_ALIGN_OF_DOUBLE  4
+#define PR_ALIGN_OF_POINTER 4
+#define PR_ALIGN_OF_WORD    4
+
+#define PR_BYTES_PER_WORD_LOG2   2
+#define PR_BYTES_PER_DWORD_LOG2  3
+
 #else
 
 #error "Unknown CPU architecture"
diff -r 8487beb203e9 -r ab7d5c504daf nspr/pr/include/md/_linux.h
--- a/nspr/pr/include/md/_linux.h	Tue Sep 03 11:25:11 2019 +0200
+++ b/nspr/pr/include/md/_linux.h	Tue Sep 03 11:33:15 2019 +0200
@@ -65,6 +65,8 @@
 #define _PR_SI_ARCHITECTURE "arc"
 #elif defined(__nios2__)
 #define _PR_SI_ARCHITECTURE "nios2"
+#elif defined(__microblaze__)
+#define _PR_SI_ARCHITECTURE "microblaze"
 #else
 #error "Unknown CPU architecture"
 #endif
